crates/hyprstream/src/cli/git_handlers.rs

```diff
@@ -1014,21 +946,20 @@ pub async fn handle_infer(
         info!("Generated {} tokens", result.split_whitespace().count());
     } else {
         // Generate all at once using streaming with collection
-        let mut response = String::new();
+        let response = std::sync::Arc::new(std::sync::Mutex::new(String::new()));
+        let response_clone = response.clone();
         engine
-            .generate_streaming_with_params(
-                &request.prompt,
-                request.max_tokens,
-                request.temperature,
-                request.top_p,
-                request.top_k,
-                request.repeat_penalty,
-                |token| {
-                    response.push_str(token);
+            .generate_streaming(
+                request.clone(),
+                move |token| {
+                    if let Ok(mut r) = response_clone.lock() {
```

Comment:
```
let response_clone = response_clone.clone();
                    // Spawn a lightweight task to avoid blocking the generator callback
                    tokio::spawn(async move {
                        if let Ok(mut r) = response_clone.lock().await {
                            r.push_str(token);
                        } else {
                            warn!("response buffer lock poisoned");
                        }
                    });
```